webapp/src/components/CurrencyTable.jsx

```diff
@@ -29,33 +31,72 @@ export const CurrencyTable = () => {
   // Updates selected currency in global state and closes modal
   function handleCurrencyChange(newTokenIndex) {
 
-    // Find URL of token logo, symbol, and type
+    // Find URL of token logo, symbol, type, and address
     let newImageUrl = findCurrencyLogoUrl(newTokenIndex, tokenListState.state.tokenList);
     let newSymbol = tokenListState.state.tokenList.tokens[newTokenIndex].symbol;
     let newType = tokenListState.state.tokenList.tokens[newTokenIndex].type;
+    let newAddress = tokenListState.state.tokenList.tokens[newTokenIndex].address;
 
     // Find correct input to update
     switch (inputs.state.currencySelectionModal.selectedInput) {
       case 'from':
         dispatch({ type: 'UPDATE_FROM_SELECTED_CURRENCY',
-          payload: { tokenIndex: newTokenIndex, logoUrl: newImageUrl, symbol: newSymbol, type: newType }
+          payload: { 
+            tokenIndex: newTokenIndex, 
+            logoUrl: newImageUrl, 
+            symbol: newSymbol, 
+            type: newType, 
+            address: newAddress }
         });
+
+        // Update allowance of from token
+        (async function () { 
+          if (newType == "NEP-21") {
+            await delay(1000).then(async function() {
+              try {
+                let allowance = await getAllowance(inputs.state.swap.from);
+                dispatch({ type: 'UPDATE_FROM_ALLOWANCE', payload: { allowance: allowance } });
+              } catch (e) {
+                console.error(e);
+              }
+            });
+          }
+        })();
+
         break;
       case 'to':
         dispatch({ type: 'UPDATE_TO_SELECTED_CURRENCY',
-          payload: { tokenIndex: newTokenIndex, logoUrl: newImageUrl, symbol: newSymbol, type: newType }
+          payload: { 
+            tokenIndex: newTokenIndex, 
+            logoUrl: newImageUrl, 
+            symbol: newSymbol, 
+            type: newType, 
+            address: newAddress }
         });
         break;
       case 'input1':
         dispatch({ type: 'UPDATE_INPUT1_SELECTED_CURRENCY',
-          payload: { tokenIndex: newTokenIndex, logoUrl: newImageUrl, symbol: newSymbol, type: newType }
+          payload: { 
+            tokenIndex: newTokenIndex, 
+            logoUrl: newImageUrl, 
+            symbol: newSymbol, 
+            type: newType, 
+            address: newAddress }
         });
         break;
       case 'input2':
         dispatch({ type: 'UPDATE_INPUT2_SELECTED_CURRENCY',
-          payload: { tokenIndex: newTokenIndex, logoUrl: newImageUrl, symbol: newSymbol, type: newType }
+          payload: { 
+            tokenIndex: newTokenIndex, 
+            logoUrl: newImageUrl, 
+            symbol: newSymbol, 
+            type: newType, 
+            address: newAddress }
         });
```

Comment:
the `payload` is the same everywhere. Let's create it once before the switch, and reuse it.